yarn-project/end-to-end/src/utils.ts

```diff
@@ -96,28 +128,43 @@ export async function setup(numberOfAccounts = 1): Promise<{
 
     const contractDeployer = new ContractDeployer(SchnorrAccountContractAbi, aztecRpcServer, publicKey);
     const deployMethod = contractDeployer.deploy();
-    const tx = deployMethod.send({ contractAddressSalt: salt });
-    await tx.isMined(0, 0.1);
-    const receipt = await tx.getReceipt();
+    await deployMethod.simulate({ contractAddressSalt: salt });
+    txContexts.push({
+      tx: undefined,
+      deployMethod,
+      salt,
+      privateKey,
+      deploymentData,
+    });
+  }
+
+  for (const context of txContexts) {
```

Comment:
why does this need to be assigned separately? (just curious)

Reply:
It's the call to 'send' that is important. By breaking this up into 3 loops it creates/simulates all contract deployments, then sends all the transactions together before waiting for them to settle. This aims to get all of the transactions into the same rollup and speed the test up.